### PR TITLE
Introduce Videopress Core JS Package

### PR DESCRIPTION
**Pull request by pgk**
Creates the VideoPress Core package skeleton. 

The overall idea is to reuse existing [VideoPress JS library code](https://github.com/Automattic/jetpack/tree/trunk/projects/packages/videopress/src/client/lib/resumable-file-uploader) outside of the Jetpack/VideoPress codebase (for example, in themes built on top of Jetpack/VideoPress or integrations).

As a next step, the resumable uploader and the other related parts of the code will be moved to this package.
The uploader code "as-is", needs [get-media-token](https://github.com/Automattic/jetpack/tree/trunk/projects/packages/videopress/src/client/lib/get-media-token) and some other TS types.

Eventually, this would:

* Be deployed as an npm package.
* Contain/replace [videopress/lib](https://github.com/Automattic/jetpack/tree/trunk/projects/packages/videopress/src/client/lib).

## Proposed changes:

* None at this point.

## Jetpack product discussion

pxWta-1PG-p2

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

* At this point this PR contains only the skeleton generated from `jetpack generate`, so I guess not much functionality to test. But maybe I missed something important on Generate.
* the build command should "work": `pnpm jetpack build --deps js-packages/videopress-core`

### CODE REVIEW

File: projects/js-packages/videopress-core/jsconfig.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// Config for a JS-using project. If using typescript, rename this to "tsconfig.json".
+{
+	"compilerOptions": {
+		"jsx": "react"
+	},
+	// List all sources and source-containing subdirs.
+	"include": [ "./index.jsx", "./src" ]
```
Comment (dhasilva) [inline projects/js-packages/videopress-core/jsconfig.json:7]: There are many TS files in VideoPress, so this probably needs to be `tsconfig.json`

**Comment by pgk** [comment]
Getting this on build, I'll see about updating my local pnpm https://github.com/orgs/pnpm/discussions/4847

File: projects/js-packages/videopress-core/composer.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+		{
+			"type": "path",
+			"url": "../../packages/*",
+			"options": {
+				"monorepo": true
+			}
+		}
+	],
+	"minimum-stability": "dev",
+	"prefer-stable": true
```
Comment (jeherve) [inline projects/js-packages/videopress-core/composer.json]: I created the mirror repo, so we now can add this to our project:

```suggestion
	"prefer-stable": true,
	"extra": {
		"autotagger": true,
		"npmjs-autopublish": true,
		"mirror-repo": "Automattic/jetpack-videopress-core",
		"changelogger": {
			"link-template": "https://github.com/Automattic/jetpack-videopress-core/compare/${old}...${new}"
		}
	}
```

**Review by jeherve — CHANGES_REQUESTED** [review]

File: projects/js-packages/videopress-core/composer.json (hunk 2 above)
Comment (pgk) [reply projects/js-packages/videopress-core/composer.json]: Thanks, done!

**Review by jeherve — APPROVED** [review]
Let's merge this now, and iterate in follow-up PRs. 👍

**Pull request merged**